modules/test/playwright/tests/style-book-web/previewStyleBookOnDisplayPageTemplates.spec.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Page, expect, mergeTests} from '@playwright/test';
+
+import {isolatedSiteTest} from '../../fixtures/isolatedSiteTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {pageEditorPagesTest} from '../../fixtures/pageEditorPagesTest';
+import {styleBookPageTest} from '../../fixtures/styleBookPageTest';
+import {workflowPagesTest} from '../../fixtures/workflowPagesTest';
+import {PageEditorPage} from '../../pages/layout-content-page-editor-web/PageEditorPage';
+import {DisplayPageTemplatesPage} from '../../pages/layout-page-template-admin-web/DisplayPageTemplatesPage';
+import {StyleBooksPage} from '../../pages/style-book-web/StyleBooksPage';
+import getRandomString from '../../utils/getRandomString';
+import {blogsPagesTest} from '../blogs-web/fixtures/blogsPagesTest';
+
+const test = mergeTests(
+	blogsPagesTest,
+	isolatedSiteTest,
+	loginTest(),
+	pageEditorPagesTest,
+	styleBookPageTest,
+	workflowPagesTest
+);
+
+const changeStyleAndCheckPreview = async (
+	colorHEX: string,
+	colorRGB: string,
+	displayPageTemplatesPage: DisplayPageTemplatesPage,
+	page: Page,
+	pageEditorPage: PageEditorPage,
+	site: Site,
+	styleBooksPage: StyleBooksPage,
+	displayPageTemplateName: string
+) => {
+	await styleBooksPage.goto(site.friendlyUrlPath);
+	await styleBooksPage.editStyleBook('Test Style Book Name');
+
+	await page.getByRole('button', {name: 'Color System'}).click();
+	await page.getByRole('menuitem', {name: 'General'}).click();
+
+	await styleBooksPage.updateTokenInputColor('Body Color', colorHEX);
+
+	expect(page.getByText('Saved')).toBeVisible();
+
+	await styleBooksPage.publishStyleBook();
+
+	await displayPageTemplatesPage.goto(site.friendlyUrlPath);
+
+	await displayPageTemplatesPage.editTemplate(displayPageTemplateName);
```

Comment:
This is not actually testing the same thing as the original test. The original test is testing the preview functionality of style books. So we shouldn't be navigating to the display page template to test the preview. We should be using the preview functionality in Style Books directly.

<img width="1715" alt="Screenshot 2024-09-09 at 10 30 24 AM" src="https://github.com/user-attachments/assets/5415399d-f4f5-486b-ba73-67cdfd2042ff">

cc: @thiago-buarqque

Reply:
Please review all the old tests to make sure we are accurately reproducing them in playwright.

Reply:
@ethib137, this is a preview to check and confirm the color change. 
You are right, I wasn't testing the preview on styleBook so I added a test only to create a styleBook and another to check the display page template preview on the stylebook.

Reply:
This looks odd. Can you reproduce this manually or only in the automated test?

Reply:
Manually it doesn't let me edit the page and there's a `cursor: not-allowed`. But I can see the heading.

Reply:
What's the reason for not wanting to test the heading color?

Reply:
Because they are not accessible on playwright, I cannot see the heading color.

Reply:
If you want me to take a look and try to play with it locally, let me know.